zyte_common_items/components/metadata.py

```diff
@@ -0,0 +1,83 @@
+from datetime import datetime
+from typing import Dict, List, Optional, Type, TypeVar
+
+import attrs
+
+from zyte_common_items import Item
+from zyte_common_items._dateutils import parse_iso_datetime
+from zyte_common_items.util import convert_to_class
+
+MetadataT = TypeVar("MetadataT", bound="BaseMetadata")
+
+
+class BaseMetadata(Item):
+    def cast(self, cls: Type[MetadataT]) -> MetadataT:
+        """Convert *value*, a metadata instance, into a different metadata *cls*."""
+        return convert_to_class(self, cls)
```

Comment:
I've moved it to a method because it was not clear where to put the function otherwise; this casting is not a converter, and not a processor, and it's specific to metadata.

The disadvantage (or, maybe, a difference), as compared to the previous aproach, is that previously to convert something the first argument didn't have to be a metadata subclass, i.e. other classes, outside of our hierarchy, were supported. But I haven't came up with a use case for this.